crates/red_knot_python_semantic/resources/mdtest/type_api.md

```diff
@@ -432,3 +432,21 @@ def _(
     reveal_type(c2)  # revealed: () -> int
     reveal_type(c3)  # revealed: (x: int, y: str) -> None
 ```
+
+## Invalid
+
+```py
+from knot_extensions import TypeOf
+
+# error: [invalid-type-form] "`knot_extensions.TypeOf` requires exactly one argument when used in a type expression"
+def f(x: TypeOf) -> None:
+    reveal_type(x)  # revealed: Unknown
+```
+
+```py
+from knot_extensions import CallableTypeFromFunction
+
+# error: [invalid-type-form] "`knot_extensions.CallableTypeFromFunction` requires exactly one argument when used in a type expression"
+def f(x: CallableTypeFromFunction) -> None:
+    reveal_type(x)  # revealed: Unknown
+```
```

Comment:
I think it might make sense to include these in their respective section as they already contains some invalid form:
* `TypeOf`: https://github.com/MatthewMckee4/ruff/blob/make-in-type-expression-exhaustive/crates/red_knot_python_semantic/resources/mdtest/type_api.md#typeof
* `CallableTypeFromFunction`: https://github.com/MatthewMckee4/ruff/blob/make-in-type-expression-exhaustive/crates/red_knot_python_semantic/resources/mdtest/type_api.md#callabletypefromfunction